extensions/sql/data/Utils.cpp

```diff
@@ -0,0 +1,61 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Utils.h"
+
+#include <algorithm>
+#include  <cctype>
+#include  <regex>
+#include  <sstream>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+std::string toLower(const std::string& str) {
+  std::string ret;
+
+  // (int(*)(int))std::tolower - to avoid compilation error 'no matching overloaded function found'. 
+  // It is described in https://stackoverflow.com/questions/5539249/why-cant-transforms-begin-s-end-s-begin-tolower-be-complied-successfu.
+  std::transform(str.begin(), str.end(), std::back_inserter(ret), (int(*)(int))std::tolower);
```

Comment:
Given the IO limit we surely have in case of an SQL processor I think saving an allocation doesn't really mean a lot. 
Don't get me wrong, you are correct, but it wouldn't change anything.

Reply:
ok, feel free to mark this thread as resolved